### PR TITLE
More robust gen_protos on jenkins.

### PR DESCRIPTION
**Pull request by robertwb**
Don't use the globally shared build directory, which pip may refuse
to use if there's a failed install.

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @aaltay

File: sdks/python/gen_protos.py [hunk 1]
```diff
@@ -105,9 +107,13 @@ def generate_proto_files():
 def _install_grpcio_tools_and_generate_proto_files():
   install_path = os.path.join(
       os.path.dirname(os.path.abspath(__file__)), '.eggs', 'grpcio-wheels')
+  build_path = install_path + '-build'
+  if os.path.exists(build_path):
+    shutil.rmtree(build_path)
   logging.warning('Downloading a grpcio-tools to %s' % install_path)
```
Comment (aaltay) [inline sdks/python/gen_protos.py]: Downloading -> Installing

File: sdks/python/gen_protos.py [hunk 2]
```diff
@@ -105,9 +107,13 @@ def generate_proto_files():
...
   subprocess.check_call(
-      ['pip', 'install', '-t', install_path, '--upgrade', GRPC_TOOLS])
+      ['pip', 'install', '-t', install_path, '-b', build_path,
```
Comment (aaltay) [inline sdks/python/gen_protos.py]: -t -> --target

File: sdks/python/gen_protos.py (hunk 1 above)
Comment (robertwb) [inline sdks/python/gen_protos.py]: Done.

File: sdks/python/gen_protos.py (hunk 2 above)
Comment (robertwb) [inline sdks/python/gen_protos.py]: Done.

**Comment by robertwb** [comment]
Thanks.

**Comment by jkff** [comment]
This PR seems to greatly slow down `mvn clean compile` because now instead of just compiling Java code, this also installs some python packages and compiles a ton of C code.

Can still be worked around by adding `-pl '!sdks/python'`, which brings time for `mvn clean compile test-compile` from 5:30 minutes to 1:30.

**Comment by robertwb** [comment]
On most machines pre-compiled wheels should be available to avoid compiling
all the C code. (This is, in fact, why I'm avoiding setup_requires...) What
system are you using?

Even faster is simply installing grcpio_tools into your Python (it checks
to see if it's available before fetching it.)

**Pull request closed without merging**